lib/meta.nix

```diff
@@ -67,4 +67,23 @@ rec {
   */
   hiPrioSet = set: mapDerivationAttrset hiPrio set;
 
+
+  /* Check to see if a platform is matched by the given `meta.platforms`
+     element.
+
+     A `meta.platform` pattern is either
+
+       1. (legacy) a system string.
+
+       2. (modern) a pattern for the platform `parsed` field.
+
+     We can inject these into a patten for the whole of a structured platform,
+     and then match that.
+  */
+  platformMatch = platform: elem: let
```

Comment:
CC @vcunat I did this and not `matchPlatform` because the platform comes before the pattern, but maybe it's best to keep normal English word order here.